azurebatchdockerhelp.md

```diff
@@ -0,0 +1,56 @@
+## Azure Batch with docker containers
+Azure Batch has the functionality to run jobs and tasks within a docker container. Using docker containers to execute REEF jobs has the benefit of setting up the dependencies in light-weight docker containers instead of the vm node. This section describes how you can configure the pool and REEF to execute REEF jobs inside docker containers.
+
+####	1. Create Dockerfile for your OS with REEF and other dependencies 
+##### Option: Windows Docker image
+
+Following example of dockerfile targets the windowsservercode on dockerhub and the pre-requisites for java and other path variables (the reference to vcruntime140.dll is since the default Windows Server core image is missing this important dll to be able to run java programs):
+
+````
+FROM microsoft/windowsservercore:latest
+ADD http://javadl.oracle.com/webapps/download/AutoDL?BundleId=207775 c:\jre-8u91-windows-x64.exe
+RUN powershell -Command Start-Process -FilePath C:\jre-8u91-windows-x64.exe -PassThru -Wait -ArgumentList \"/s /L c:\Java64.log\"
+ENV 'JAVA_HOME' 'C:\Program Files\Java\jre1.8.0_91\'
+ENV 'PATH' 'C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Users\ContainerAdministrator\AppData\Local\Microsoft\WindowsApps;C:\Program Files\Java\jre1.8.0_91\bin\'
+ADD vcruntime140.dll C:/Windows/System32/vcruntime140.dll
+RUN del c:\jre-8u91-windows-x64.exe
+
+# Add your own dependencies here
+````
+
+##### Option: Ubuntu Docker image
+
+The following example installs the jdk and necessary utilities required by REEF:
+````
+FROM ubuntu
+RUN apt-get update && apt-get install -y default-jdk unzip
+ENV JAVA_HOME /usr/bin/java
+
+# Add your own dependencies here
+````
+
+#### 2. [Create a docker image from the dockerfile and publish the docker image to Azure Container service.](https://docs.microsoft.com/en-us/azure/container-service/kubernetes/container-service-tutorial-kubernetes-prepare-acr)
+
+#### 3. Create a Pool in the Azure Batch account for [Container workloads](https://docs.microsoft.com/en-us/azure/batch/batch-docker-container-workloads)  
+Use the following settings when creating the pool:  
```

Comment:
>Use the following settings when creating the pool:  [](start = 0, length = 51)

Should we call out anything in regards to the InboundNATPool?

Reply:
Yep. Added a note with a link.

---
In reply to: [202437796](https://github.com/sharathmalladi/reef/pull/7#discussion_r202437796) [](ancestors = 202437796)